src/manifolds/StiefelEuclideanMetric.jl

```diff
@@ -185,6 +185,55 @@ function retract_project!(M::Stiefel, q, p, X, t::Number)
     return q
 end
 
+@doc raw"""
+    Y = riemannian_Hessian(M::Stiefel, p, G, H, X)
+    riemannian_Hessian!(M::Stiefel, Y, p, G, H, X)
+
+Compute the Riemannian Hessian ``\operatorname{Hess} f(p)[X]`` given the
+Euclidean gradient ``∇ f(\tilde p)`` in `G` and the Euclidean Hessian ``∇^2 f(\tilde p)[\tilde X]`` in `H`,
+where ``\tilde p, \tilde X`` are the representations of ``p,X`` in the embedding,.
+
+Here, we adopt Eq. (5.6) [Nguyen:2023](@cite), where we use for the [`EuclideanMetric`](https://juliamanifolds.github.io/ManifoldsBase.jl/stable/manifolds.html#ManifoldsBase.EuclideanMetric)
+``α_0=α_1=1`` in their formula. Then the formula reads
+
+```math
+    \operatorname{Hess}f(p)[X]
+    =
+    \operatorname{proj}_{T_p\mathcal M}\Bigl(
+        ∇^2f(p)[X] - \frac{1}{2} X \bigl((∇f(p))^{\mathrm{H}p + p^{\mathrm{H}}∇f(p)\bigr)
+    \Bigr).
+```
```

Comment:
This formula doesn't render correctly.

Reply:
Ah I see, I am too much used to the LaTeX-VSCode closing `}` for me. It is missing after the ^H. Will add that and check the others as well